arbor/fvm_layout.hpp

```diff
@@ -68,8 +110,43 @@ struct cv_geometry {
         return cell_cv_divs.empty()? 0: cell_cv_divs.size()-1;
     }
 
-    size_type location_cv(size_type cell_idx, mlocation loc) const {
-        return cell_cv_divs.at(cell_idx)+branch_cv_map.at(cell_idx).at(loc.branch)(loc.pos).second;
+    size_type n_branch(size_type cell_idx) const {
+        return branch_cv_map.at(cell_idx).size();
+    }
+
+    size_type location_cv(size_type cell_idx, mlocation loc, cv_prefer::type prefer) const {
+        auto& pw_cv_offset = branch_cv_map.at(cell_idx).at(loc.branch);
+        auto zero_extent = [&pw_cv_offset](auto j) {
+            return pw_cv_offset.interval(j).first==pw_cv_offset.interval(j).second;
+        };
+
+        auto i = pw_cv_offset.index_of(loc.pos);
+        auto i_max = pw_cv_offset.size()-1;
+        auto cv_prox = pw_cv_offset.interval(i).first;
+
+        // index_of() should have returned right-most matching interval.
+        arb_assert(i==i_max || loc.pos<pw_cv_offset.interval(i+1).first);
+
+        using namespace cv_prefer;
+        switch (prefer) {
+        case cv_distal:
+            break;
+        case cv_proximal:
+            if (loc.pos==cv_prox && i>0) --i;
+            break;
+        case cv_nonempty:
+            if (zero_extent(i)) {
```

Comment:
Can we had adjacent `zero_extent` CVs? If so, what happens if `i` has zero-length cvs either side?

Reply:
This can only occur if we have multiple zero-length CVs at the end of a cable. The CV building algorithm will not make such discretizations, but the `location_cv` code's heuristic is: give the nonempty CV if there is one, otherwise what we have at hand.